haxepunk/graphics/hardware/DrawCommand.hx

```diff
@@ -154,26 +153,45 @@ class DrawCommand
 
 	function new() {}
 
-	public inline function match(texture:BitmapData, shader:Shader, smooth:Bool, blend:BlendMode, clipRect:Rectangle):Bool
+	/**
+	 * Compares values to this draw command to see if they all match. This is used by the batcher to reuse the previous draw command.
+	 */
+	public inline function match(texture:Texture, shader:Shader, smooth:Bool, blend:BlendMode, clipRect:Rectangle):Bool
```

Comment:
I'm surprised you're not seeing "can't inline a non-final return" here since this has many possible return points. Wonder if Haxe improved this recently, or if I just misunderstand how that works. In the hxcpp output, does this actually get inlined?